src/miner_lora.erl

```diff
@@ -0,0 +1,399 @@
+-module(miner_lora).
+
+-behaviour(gen_server).
+
+-export([
+    start_link/1,
+    send/1,
+    send_poc/5
+]).
+
+-export([
+    init/1,
+    handle_call/3,
+    handle_cast/2,
+    handle_info/2,
+    terminate/2,
+    code_change/3
+]).
+
+-include_lib("helium_proto/src/pb/blockchain_state_channel_v1_pb.hrl").
+-include("lora.hrl").
+
+-record(gateway, {
+    mac,
+    ip,
+    port,
+    sent =0,
+    received = 0,
+    dropped = 0,
+    status,
+    rtt_samples = [],
+    rtt=5000000 %% in microseconds
+}).
+
+-record(state, {
+    socket,
+    gateways = #{}, %% keyed by MAC
+    packet_timers = #{}, %% keyed by token
+    sig_fun,
+    pubkey_bin
+}).
+
+-type state() :: #state{}.
+-type gateway() :: #gateway{}.
+-type helium_packet() :: #helium_packet_pb{}.
+
+%% ------------------------------------------------------------------
+%% API Function Definitions
+%% ------------------------------------------------------------------
+
+start_link(Args) ->
+    gen_server:start_link({local, ?MODULE}, ?MODULE, Args, []).
+
+-spec send(helium_packet()) -> ok | {error, any()}.
+send(#helium_packet_pb{payload=Payload, timestamp=When, signal_strength=Power, frequency=Freq, datarate=DataRate}) ->
+    gen_server:call(?MODULE, {send, Payload, When, Freq, DataRate, Power, true}, 11000).
+
+-spec send_poc(binary(), any(), float(), iolist(), any()) -> ok | {error, any()}.
+send_poc(Payload, When, Freq, DataRate, Power) ->
+    gen_server:call(?MODULE, {send, Payload, When, Freq, DataRate, Power, false}, 11000).
+
+%% ------------------------------------------------------------------
+%% gen_server Function Definitions
+%% ------------------------------------------------------------------
+
+init(Args) ->
+    UDPIP = maps:get(radio_udp_bind_ip, Args),
```

Comment:
Expecting that these keys are always there or the miner_lora process blows up I presume...